lib/auth/helpers.go

```diff
@@ -227,6 +228,14 @@ func NewTestAuthServer(cfg TestAuthServerConfig) (*TestAuthServer, error) {
 	access := local.NewAccessService(srv.Backend)
 	identity := local.NewIdentityService(srv.Backend)
 
+	emitter, err := events.NewCheckingEmitter(events.CheckingEmitterConfig{
```

Comment:
Introducing the CheckingEmitter breaks a few tests which I have fixed, but I feel like this was the appropriate course of action because frankly it was very unexpected the way that file audit log behaved without this. Events were written to a log based on today's date, but with a `Time` of 0001-01-01T00:00:00, which causes very weird behaviour when trying to search the event log. Now, they are appropriately written to a log file based on the fake test clocks date, with a time that matches the fake test clock. This seems more consistent with an engineers expectations.